cscs-checks/libraries/io/netcdf_compile_run.py

```diff
@@ -35,6 +45,25 @@ def __init__(self, lang, linkage):
         self.tags = {'production'}
 
     def setup(self, partition, environ, **job_opts):
+        if (environ.name == 'PrgEnv-cray-nompi'):
+            self.modules = ['netcdf/4.4.1.1-gmvolf-17.02',
+                            'netcdf-c++/4.3.0-gmvolf-17.02',
+                            'netcdf-fortran/4.4.4-gmvolf-17.02'
+                           ]
+            self.build_system.ldflags = ['-I$EBROOTNETCDF/include',
+                                         '-I$EBROOTNETCDFMINCPLUSPLUS/include',
+                                         '-I$EBROOTNETCDFMINFORTRAN/include',
+                                         '-L$EBROOTNETCDF/lib',
+                                         '-L$EBROOTNETCDFMINCPLUSPLUS/lib',
+                                         '-L$EBROOTNETCDFMINFORTRAN/lib',
+                                         '-L$EBROOTNETCDF/lib64',
+                                         '-L$EBROOTNETCDFMINCPLUSPLUS/lib64',
+                                         '-L$EBROOTNETCDFMINFORTRAN/lib64',
+                                         '-lnetcdf',
+                                         '-lnetcdf_c++4',
+                                         '-lnetcdff'
+                                        ]
```

Comment:
This is a clear indication that the `PrgEnv-cray-nompi` and also the `netcdf` modules is not properly setup. Have you tried setting the corresponding variables inside the definition of the programming environment in ReFrame.